src/layout/node.ts

```diff
@@ -74,36 +82,58 @@ export default function(node: Node, source: Source): void {
             if (element.namespaceURI === Constant.SVG_NAMESPACE) { tag = Constant.SVG_PREFIX + tag; }
 
             switch (tag) {
+                case "HTML":
+                    // Skip processing nodes within iframe if they were already removed from the DOM before we got to processing them.
+                    // In those cases, we skip them and continue processing rest of the nodes.
+                    let parentFrame = insideFrame && parent ? dom.iframe(parent) : null;
+                    if (insideFrame && parentFrame === null) { break; }
+                    let htmlData = { tag, attributes: getAttributes(element.attributes) };
+                    dom[call](node, parentFrame ? parentFrame : parent, htmlData, source);
+                    break;
                 case "SCRIPT":
                 case "NOSCRIPT":
                 case "META":
                     break;
                 case "HEAD":
                     let head = { tag, attributes: getAttributes(element.attributes) };
                     // Capture base href as part of discovering DOM
-                    head.attributes[Constant.BASE_TAG] = location.protocol + "//" + location.hostname;
+                    if (insideFrame === false) { head.attributes[Constant.BASE_ATTRIBUTE] = location.protocol + "//" + location.hostname; }
                     dom[call](node, parent, head, source);
                     break;
                 case "STYLE":
                     let attributes = getAttributes(element.attributes);
                     let styleData = { tag, attributes, value: getStyleValue(element as HTMLStyleElement) };
                     dom[call](node, parent, styleData, source);
                     break;
+                case "IFRAME":
+                    let iframe = node as HTMLIFrameElement;
+                    let frameData = { tag, attributes: getAttributes(iframe.attributes) };
+                    if (dom.sameorigin(iframe)) {
+                        mutation.monitor(iframe);
+                        frameData.attributes[Constant.SAME_ORIGIN_ATTRIBUTE] = "true";
+                        if (iframe.contentDocument && iframe.contentWindow && iframe.contentDocument.readyState !== "loading") {
```

Comment:
>readyState [](start = 101, length = 10)

Is the idea if the state is loading then you defer scanning content of the iframe until you get the load event that you have listened to? Wouldn't that miss mutations as the iframe is in the process of loading?

Reply:
Yes, we will miss the mutations but when we get to processing nodes within the IFRAME, we will get the current state at the point. This is true for the uber DOM too. If Clarity wires up late or we have to wait (due to async nature) and mutations happen inbetween, we merge them into initial discovery.